src/common/OutputDataSocket.cc

```diff
@@ -299,8 +299,7 @@ int OutputDataSocket::dump_data(int fd)
 {
   m_lock.Lock(); 
   list<bufferlist> l;
-  l = data;
-  data.clear();
+  l.swap(data);
```

Comment:
@joscollin you can try to implement `swap()`, and see what the difference it is.